internal/controller/mantlebackupconfig_controller_test.go

```diff
@@ -0,0 +1,454 @@
+type ObjectConstraint[T any] interface {
+	client.Object
+	*T
+}
+
+func checkCreatedEventually[T any, OC ObjectConstraint[T]](ctx context.Context, name, namespace string) {
+	var obj T
+	Eventually(func() error {
+		err := k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, OC(&obj))
+		if err != nil {
+			return err
+		}
+		return nil
+	}).Should(Succeed())
+}
+
+func checkDeletedEventually[T any, OC ObjectConstraint[T]](ctx context.Context, name, namespace string) {
+	var obj T
+	Eventually(func() error {
+		err := k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, OC(&obj))
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("\"%s\" is not deleted yet", name)
+	}).Should(Succeed())
+}
+
+var freshNameIndex = 0
+
+func getFreshName(prefix string) string {
+	freshNameIndex++
+	return fmt.Sprintf("%s-%d", prefix, freshNameIndex)
+}
+
+var _ = Describe("MantleBackupConfig controller", func() {
+	ctx := context.Background()
+	var stopFunc func()
+	errCh := make(chan error)
+
+	var reconciler *MantleBackupConfigReconciler
+	scheme := runtime.NewScheme()
+
+	storageClassClusterID := dummyStorageClassClusterID
+	storageClassName := dummyStorageClassName
+
+	BeforeEach(func() {
+		err := batchv1.AddToScheme(scheme)
+		Expect(err).NotTo(HaveOccurred())
+		err = mantlev1.AddToScheme(scheme)
+		Expect(err).NotTo(HaveOccurred())
+
+		mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+			Scheme: scheme,
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		reconciler = NewMantleBackupConfigReconciler(k8sClient, mgr.GetScheme(), storageClassClusterID, "0s")
+		err = reconciler.SetupWithManager(mgr)
+		Expect(err).NotTo(HaveOccurred())
+
+		executeCommand = mockExecuteCommand
+
+		ctx, cancel := context.WithCancel(ctx)
+		stopFunc = cancel
+		go func() {
+			errCh <- mgr.Start(ctx)
+		}()
+		time.Sleep(100 * time.Millisecond)
+	})
+
+	AfterEach(func() {
+		stopFunc()
+		Expect(<-errCh).NotTo(HaveOccurred())
+	})
+
+	DescribeTable("MantleBackupConfigs with correct fields",
+		func(schedule, expire string) {
+			ns := createNamespace()
+			pvName := getFreshName("pv")
+			pvcName := getFreshName("pvc")
+
+			err := createBoundPVC(ctx, ns, pvName, pvcName, storageClassName)
+			Expect(err).NotTo(HaveOccurred())
+
+			mbc := mantlev1.MantleBackupConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      getFreshName("mbc"),
+					Namespace: ns,
+				},
+				Spec: mantlev1.MantleBackupConfigSpec{
+					PVC:      pvcName,
+					Schedule: schedule,
+					Expire:   expire,
+					Suspend:  false,
+				},
+			}
+			err = k8sClient.Create(ctx, &mbc)
+			Expect(err).NotTo(HaveOccurred())
+		},
+		Entry("various expires 1", "0 0 * * *", "24h"),
+		Entry("various expires 1", "0 0 * * *", "1d"),
```

Comment:
```suggestion
		Entry("various expires 1", "0 0 * * *", "1d"),
```
```suggestion
		Entry("various expires 2", "0 0 * * *", "1d"),
```

Subsequent 1st argus should also be updated.